src/custom/hooks/useSwapCallback.ts

```diff
@@ -2,10 +2,121 @@ import { SwapCallbackState } from '@src/hooks/useSwapCallback'
 import { useMemo } from 'react'
+import useTransactionDeadline from '@src/hooks/useTransactionDeadline'
+import { BigNumber } from 'ethers'
+import { useAddPendingOrder } from '../state/operator/hooks'
+import { AddPendingOrderParams, OrderCreation, OrderID, OrderKind, OrderStatus } from '../state/operator/actions'
+import { isAddress, shortenAddress } from '@src/utils'
+
+interface PostOrderParams {
+  account: string
+  chainId: ChainId
+  trade: Trade
+  validTo: number
+  recipient: string
+  recipientAddressOrName: string | null
+  addPendingOrder: (order: AddPendingOrderParams) => void
+}
+
+const MAX_VALID_TO_EPOCH = BigNumber.from('0xFFFFFFFF').toNumber() // Max uint32 (Feb 07 2106 07:28:15 GMT+0100)
+const DEFAULT_APP_ID = 0
+
+type UnsignedOrder = Omit<OrderCreation, 'signature'>
+
+function getSummary(params: PostOrderParams): string {
+  const { trade, account, recipient, recipientAddressOrName } = params
+
+  const inputSymbol = trade.inputAmount.currency.symbol
+  const outputSymbol = trade.outputAmount.currency.symbol
+  const inputAmount = trade.inputAmount.toSignificant(3)
+  const outputAmount = trade.outputAmount.toSignificant(3)
+
+  const base = `Swap ${inputAmount} ${inputSymbol} for ${outputAmount} ${outputSymbol}`
+  const withRecipient =
+    recipient === account
+      ? base
+      : `${base} to ${
+          recipientAddressOrName && isAddress(recipientAddressOrName)
+            ? shortenAddress(recipientAddressOrName)
+            : recipientAddressOrName
+        }`
+
+  return withRecipient
+}
+
+function signOrder(unsignedOrder: UnsignedOrder): string {
+  console.log('TODO: Sign order', unsignedOrder)
+  // TODO: Mocked, next PR
+  return '0xd6741f8031e7a7ea70f1b6c14a71e9d0d7d5aae54c157d8368ad4cabd7279a363955fa169a634469da01010bfefbacc2c1e2e7aaeb0c42049192e6359ed572281b'
+}
+
+async function postOrderApi(params: PostOrderParams, signature: string): Promise<OrderID> {
+  const { inputAmount } = params.trade
+  // TODO: Pretend we call the API
+  console.log('TODO: call API and include the signature', signature)
+  return new Promise<string>(async (resolve, reject) => {
+    setTimeout(() => {
+      if (inputAmount.toExact() === '0.1') {
+        // Force error for testing
+        console.log('[useSwapCallback] Ups, we had a small issue!')
+        reject(new Error('Mock error: The flux capacitor melted'))
+      } else {
+        // Pretend all went OK
+        console.log('[useSwapCallback] Traded successfully!')
+        resolve('123456789')
+      }
+    }, 3000)
+  })
+}
+
+async function postOrder(params: PostOrderParams): Promise<string> {
+  const { addPendingOrder, chainId, trade, validTo, account } = params
+  const { inputAmount, outputAmount } = trade
+  const [selToken] = trade.route.path
+  const [, buyToken] = trade.route.path
+
+  // Prepare order
+  const summary = getSummary(params)
+  const unsignedOrder: UnsignedOrder = {
+    sellToken: selToken.address,
+    buyToken: buyToken.address,
```

Comment:
ok for now as is, but api expects without `0x` prefix

Reply:
Yep, I'll change later then